File: src/antares_cli/agent/execution_policy.py
"""Shared execution policy for every Antares agent entry point."""

from __future__ import annotations

DEFAULT_TERMINAL_CALL_BUDGET = 15
MIN_TERMINAL_CALL_BUDGET = 1
MAX_TERMINAL_CALL_BUDGET = 50


def resolve_terminal_call_budget(value: int | None) -> int:
    """Return the terminal-call budget used by all invocation surfaces."""
    if value is None:
        return DEFAULT_TERMINAL_CALL_BUDGET
    if not MIN_TERMINAL_CALL_BUDGET <= value <= MAX_TERMINAL_CALL_BUDGET:
        raise ValueError(
            "Terminal call budget must be between "
            f"{MIN_TERMINAL_CALL_BUDGET} and {MAX_TERMINAL_CALL_BUDGET}."
        )
    return value

File: vscode-extension/src/antares/agent/executionPolicy.ts
// Port of antares_cli/agent/execution_policy.py — terminal-call budget policy.

export const DEFAULT_TERMINAL_CALL_BUDGET = 15;
export const MIN_TERMINAL_CALL_BUDGET = 1;
export const MAX_TERMINAL_CALL_BUDGET = 50;

export function resolveTerminalCallBudget(value: number | null | undefined): number {
  if (value === null || value === undefined) {
    return DEFAULT_TERMINAL_CALL_BUDGET;
  }
  if (
    !Number.isInteger(value) ||
    value < MIN_TERMINAL_CALL_BUDGET ||
    value > MAX_TERMINAL_CALL_BUDGET
  ) {
    throw new Error("Terminal call budget must be between 1 and 50.");
  }
  return value;
}
